recipes/virtual-libjpeg/all/conanfile.py

```diff
@@ -0,0 +1,57 @@
+from conan import ConanFile
+
+required_conan_version = ">=1.52.0"
+
+
+class VirtualLibjpegConan(ConanFile):
+    name = "virtual-libjpeg"
+    description = "Proxy recipe to select either libjpeg, libjpeg-turbo or mozjpeg"
+    license = "MIT"
+    topics = ("proxy", "virtual", "jpeg", "libjpeg", "libjpeg-turbo", "mozjepg")
+    homepage = "https://github.com/conan-io/conan-center-index"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "implementation": ["libjpeg", "libjpeg-turbo", "mozjpeg"],
+    }
+    default_options = {
+        "implementation": "libjpeg-turbo",
+    }
+
+    @property
+    def _impl_version(self):
+        return {
+            "libjpeg": "9e",
+            "libjpeg-turbo": "2.1.4",
+            "mozjpeg": "4.1.1",
+        }[str(self.options.implementation)]
+
+    def layout(self):
+        pass
+
+    def requirements(self):
+        self.requires(f"{self.options.implementation}/{self._impl_version}", transitive_headers=True, transitive_libs=True)
+
+    def package_id(self):
+        self.info.clear()
```

Comment:
I think you need to clear everything but options.

Reply:
You mean as a workaround for https://github.com/conan-io/conan-center-index/pull/14814#issuecomment-1357586732?

Reply:
The main issue is what I've mentioned in https://github.com/conan-io/conan/issues/12374:

> I see one main issue with such virtual recipe and current conan behavior: if a recipeA depends on this virtual recipe, the underlying recipe implementation is not anymore a direct dependency of recipeA, so I suspect package id issues.
How to design this virtual recipe so that any recipe having it as a direct dependency would behave as if either libjpeg, libjpeg-turbo or mozjpeg would be considered a direct dependency?